cmd/kubeadm/app/cmd/phases/init/addons.go

```diff
@@ -42,6 +45,12 @@ var (
 
 // NewAddonPhase returns the addon Cobra command
 func NewAddonPhase() workflow.Phase {
+	dnsLocalFlags := pflag.NewFlagSet(options.PrintManifest, pflag.ContinueOnError)
+	dnsLocalFlags.BoolVar(&dnsaddon.PrintManifestFlag, options.PrintManifest, dnsaddon.PrintManifestFlag, "Print the addon manifests to STDOUT instead of installing them")
+
+	proxyLocalFlags := pflag.NewFlagSet(options.PrintManifest, pflag.ContinueOnError)
+	proxyLocalFlags.BoolVar(&proxyaddon.PrintManifestFlag, options.PrintManifest, proxyaddon.PrintManifestFlag, "Print the addon manifests to STDOUT instead of installing them")
+
```

Comment:
If we add the flag like this, the `all` will not support this flag.
```
[root@daocloud ~]# ./kubeadm init phase addon  all --print-manifest
unknown flag: --print-manifest
To see the stack trace of this error execute with --v=5 or higher
```

Reply:
I think it's ok to not support it for "all". We could do it but it introduce some minor code complexity. On the other hand users can just execute the separate addon phases with the flag.

Reply:
https://github.com/kubernetes/kubernetes/pull/109995/files#diff-872fa7d2e3a260ba4ae68c07c622fbde9f4202d8bb0d1830103c9dbf9ab3ac05R51


If so, the global variable `var PrintManifestFlag bool = false` across the packages above is easy to implement. 
/lgtm